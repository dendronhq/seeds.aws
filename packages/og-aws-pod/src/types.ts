import { PodOptsV2, Pod, PodConfigEntry } from "./base";

export interface PodClassEntryV2 {
  id: string;
  description: string;
  kind: PodKind;
  config: () => PodConfigEntry[];
  new (opts: PodOptsV2): Pod;
}

export type PodItem = {
  id: string;
  description: string;
  podClass: PodClassEntryV2;
};

export type PodKind = "import" | "export";
