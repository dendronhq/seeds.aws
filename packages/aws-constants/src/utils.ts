import * as _ from "lodash";
import { ServiceNames, ServiceNamesNoPrefix } from "./services";

const ALIASES = {
  ACM: ServiceNames.AWS_CERTIFICATE_MANAGER,
  ALB: ServiceNames.APPLICATION_LOAD_BALANCER,
  CE: ServiceNames.AWS_COST_EXPLORER,
  CLB: ServiceNames.CLASSIC_LOAD_BALANCER,
  DMS: ServiceNames.AWS_DATABASE_MIGRATION_SERVICE,
  DS: ServiceNames.AWS_DIRECTORY_SERVICE,
  EBS: ServiceNames.AMAZON_ELASTIC_BLOCK_STORE,
  ECR: ServiceNames.AMAZON_ELASTIC_CONTAINER_REGISTRY,
  ECS: ServiceNames.AMAZON_ELASTIC_CONTAINER_SERVICE,
  EFS: ServiceNames.AMAZON_ELASTIC_FILE_SYSTEM,
  EKS: ServiceNames.AMAZON_ELASTIC_CONTAINER_SERVICE_FOR_KUBERNETES,
  ELB: ServiceNames.ELASTIC_LOAD_BALANCING,
  EMR: ServiceNames.AMAZON_ELASTIC_MAP_REDUCE,
  ES: ServiceNames.AMAZON_ELASTICSEARCH_SERVICE,
  MGH: ServiceNames.AWS_MIGRATION_HUB,
  IAM: ServiceNames.AWS_IDENTITY_ACCESS_MANAGEMENT,
  KMS: ServiceNames.AWS_KEY_MANAGEMENT_SERVICE,
  QLDB: ServiceNames.AMAZON_QUANTUM_LEDGER_DATABASE,
  S3: ServiceNames.AMAZON_SIMPLE_STORAGE_SERVICE,
  SES: ServiceNames.AMAZON_SIMPLE_EMAIL_SERVICE,
  SFTP: ServiceNames.AWS_TRANSFER_FOR_SFTP,
  SMS: ServiceNames.AWS_SERVER_MIGRATION_SERVICE,
  SNS: ServiceNames.AMAZON_SIMPLE_NOTIFICATION_SERVICE,
  SQS: ServiceNames.AMAZON_SIMPLE_QUEUE_SERVICE,
  SSO: ServiceNames.AWS_SINGLE_SIGN_ON,
};

// === AWS Utils

export function awsAliasToFull(alias: string): string {
  const key: string = alias.toUpperCase();
  // @ts-ignore
  const longName = ALIASES[key] as string;
  if (longName) {
    return normalizeServiceName(longName, {
      snakeCase: false,
      stripPrefix: true,
      slugName: false,
      expandAlias: false,
    });
  } else {
    return alias;
  }
}

// TODO: generate chart
export const AWS_SERVICE_SUFFIX: any = {
  quicksight: "QuickSight",
};
export function awsCapitalizeServiceName(serviceName: string): string {
  const capitalized: string | undefined =
    AWS_SERVICE_SUFFIX[serviceName.toLowerCase()];
  if (capitalized) {
    return capitalized;
  } else {
    return serviceName;
  }
}

export function awsGetPrefix(serviceName: string): string {
  const match = serviceName.match(/^(AWS|AMAZON)/i);
  if (!_.isNull(match)) {
    return _.trim(match[0]);
  } else {
    return "";
  }
}

export const AWS_ONE_OFF_MAP: any = {
  glacier: "S3 Glacier",
  elasticsearch: "Elasticsearch Service",
  mediastore: "Elemental MediaStore",
  medialive: "Elemental MediaLive",
  mediaconvert: "Elemental MediaConvert",
  lakeformation: "Lake Formation",
  "iot greengrass": "Greengrass",
  iotevents: "IoT Events",
  iot1click: "IoT 1-Click",
  eventbridge: "Event Bridge",
};
/** Normalize suffixes */
export function awsNormalizeOneOffName(name: string): string {
  const resp = AWS_ONE_OFF_MAP[name.toLowerCase()];
  if (!_.isUndefined(resp)) {
    return resp;
  } else {
    return name;
  }
}

export function awsSuffixToFullServiceName(suffix: string): string {
  return _.mapKeys(ServiceNames, (v) => {
    return normalizeServiceName(v as string, { stripPrefix: true });
  })[suffix] as string;
}

/**
 * Default: snake case, capitalize service name, strip paren, expand alias
 * @param name
 * @param opts
    snakeCase: false,
    stripPrefix: false,
    slugName: false,
    stripParen: true,
    expandAlias: true,
    capitalizeName: true
 */
export function normalizeServiceName(
  name: string,
  opts?: {
    snakeCase?: boolean;
    stripPrefix?: boolean;
    slugName?: boolean;
    stripParen?: boolean;
    expandAlias?: boolean;
    capitalizeName?: boolean;
  }
): string {
  const blacklist = ["S_3", "EC_2", "IO_T", "V_2", "F_S", "AP_I"];
  opts = _.defaults(opts, {
    snakeCase: false,
    stripPrefix: false,
    slugName: false,
    stripParen: true,
    expandAlias: true,
    capitalizeName: true,
  });
  let prefix = "";
  let suffix = name;
  let sep = " ";
  const match = name.match(/^(AWS|AMAZON)/i);
  if (!_.isNull(match)) {
    prefix = _.trim(match[0]);
    suffix = _.trim(name.replace(prefix, ""));
  }

  // should be before expanding
  if (opts.stripParen) {
    let paren = suffix.match(/([^(]+)(?<capture>\(.+\))/);
    if (paren && paren.groups && paren.groups.capture) {
      suffix = _.trim(suffix.replace(paren.groups.capture, ""));
    }
  }
  if (opts.expandAlias) {
    suffix = awsAliasToFull(suffix);
  }
  if (opts.capitalizeName) {
    suffix = awsCapitalizeServiceName(suffix);
  }
  suffix = awsNormalizeOneOffName(suffix);

  // destructive
  if (opts.snakeCase) {
    let tmpSuffix = _.snakeCase(suffix).toUpperCase();
    blacklist.forEach((ent) => {
      if (tmpSuffix.indexOf(ent) >= 0) {
        tmpSuffix = tmpSuffix.replace(
          new RegExp(ent, "i"),
          ent.replace("_", "")
        );
      }
    });
    suffix = tmpSuffix;
    sep = "_";
    prefix = prefix.toUpperCase();
  }

  if (opts.stripPrefix) {
    prefix = "";
  }

  if (opts.slugName) {
    suffix = suffix.replace(new RegExp(sep, "g"), "");
  }

  return _.filter([prefix, suffix], (u) => !_.isEmpty(u)).join(sep);
}

export function validateAWSServiceName(
  name: string,
  opts = { stripPrefix: false }
) {
  if (opts.stripPrefix) {
    return _.includes(_.values(ServiceNamesNoPrefix), name);
  } else {
    return _.includes(_.values(ServiceNames), name);
  }
}
