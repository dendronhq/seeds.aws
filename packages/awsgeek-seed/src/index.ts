// import { normalizeServiceName } from "@dendronhq/aws-constants/lib/src";
// import { Note } from "@dendronhq/common-all";
// import { cleanFileName } from "@dendronhq/common-server";
// import { DendronSeed, PrepareOpts } from "@dendronhq/seeds-core";
// import fs from "fs-extra";
// import path from "path";
// import remark from "remark";
// import markdownParse from "remark-parse";
// import { getOutput, ogAWSPlugin } from "./parseOgAWS";

// export default class OgAWSSeed extends DendronSeed {
//   config() {
//     return {
//       src: {
//         type: "git" as const,
//         url: "https://github.com/open-guides/og-aws.git",
//       },
//     };
//   }

//   async prepare(opts: PrepareOpts) {
//     const ctx = "prepare";
//     this.L.info({ctx, opts});
//     const { root } = opts;
//     const fpath = path.join(root, "README.md");
//     const dataPath = fs.readFileSync(fpath, { encoding: "utf8" });
//     remark()
//       .use(markdownParse, { gfm: true })
//       .use(ogAWSPlugin)
//       .processSync(dataPath);
//     const output = getOutput();
//     const dataSplit = dataPath.split("\n");
//     // skip last one, high availability
//     const notes = output.slice(0, -1).map((ent, index) => {
//       const { name, payload } = ent;
//       let start = payload.position?.start.line as number;
//       start += 1;
//       let end = output[index + 1]?.payload.position?.start.line || -1;
//       end -= 1;
//       const nname = normalizeServiceName(name);
//       const content = [`# ${nname}`].concat(dataSplit.slice(start, end));
//       const cleanName = cleanFileName(nname.toLowerCase()).replace(/,/g, "");
//       const fname = `s.${cleanName}`;
//       return new Note({
//         id: cleanName,
//         title: nname,
//         created: "0",
//         updated: "0",
//         fname,
//         body: content.join("\n"),
//         custom: {
//           source: {
//             name: "og-aws",
//             url: "https://github.com/open-guides/og-aws"
//           }
//         }
//       });
//     });
//     return notes;
//   }
// }
