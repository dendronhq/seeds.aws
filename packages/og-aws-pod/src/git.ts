import git  from 'nodegit' 
import fs  from 'fs-extra' 

export class GitRepo {
  constructor(public opts: {localUrl: string, remoteUrl: string}) {
  }

  async checkout() {
      const {localUrl, remoteUrl} = this.opts
      fs.ensureDirSync(localUrl);
      return await git.Clone.clone(remoteUrl, localUrl);
  }
}