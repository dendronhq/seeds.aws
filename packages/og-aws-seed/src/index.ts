import { Note } from "@dendronhq/common-all";
import { DendronSeed, PrepareOpts } from "@dendronhq/seeds-core";
import fs from "fs-extra";
import path from "path";

export default class OgAWSSeed extends DendronSeed {
  config() {
    return {
      src: {
        type: "git" as const,
        url: "https://github.com/AwsGeek/awsgeek.github.io",
      },
    };
  }

  async prepare(opts: PrepareOpts) {
    const ctx = "prepare";
    this.L.info({ctx, opts});
    const { root } = opts;
    const fpath = path.join(root, "images");
    const dataPath = fs.readdirSync(fpath);
    dataPath.map(ent => {
      console.log(ent);
    });
    let notes: Note[] = [];

    // const notes = output.slice(0, -1).map((ent, index) => {
    //   const { name, payload } = ent;
    //   let start = payload.position?.start.line as number;
    //   start += 1;
    //   let end = output[index + 1]?.payload.position?.start.line || -1;
    //   end -= 1;
    //   const nname = normalizeServiceName(name);
    //   const content = [`# ${nname}`].concat(dataSplit.slice(start, end));
    //   const cleanName = cleanFileName(nname.toLowerCase()).replace(/,/g, "");
    //   const fname = `s.${cleanName}`;
    //   return new Note({
    //     id: cleanName,
    //     title: nname,
    //     created: "0",
    //     updated: "0",
    //     fname,
    //     body: content.join("\n"),
    //     custom: {
    //       source: {
    //         name: "og-aws",
    //         url: "https://github.com/open-guides/og-aws"
    //       }
    //     }
    //   });
    // });
    return notes;
  }
}
