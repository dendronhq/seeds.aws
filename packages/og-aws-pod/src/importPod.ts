import { Note } from "@dendronhq/common-all";
import { cleanFileName } from "@dendronhq/common-server";
import {
  getPodPath,
  ImportConfig,
  ImportPodBaseV2,
  ImportPodOpts,
  PodConfigEntry
} from "@dendronhq/pods-core";
import fs from "fs-extra";
import path from "path";
import remark from "remark";
import markdownParse from "remark-parse";
import { URI } from "vscode-uri";
import { GitRepo } from "./git";
import { getOutput, ogAWSPlugin } from "./parseOgAWS";

type TConfig = ImportConfig & {};

export class OgAWSPod extends ImportPodBaseV2<TConfig> {
  static id: string = "pod.og-aws";
  static description: string = "import og aws content";

  static config = (): PodConfigEntry[] => {
    return [
      {
        key: "src",
        description: "where will notes be imported from",
        type: "string",
      },
    ];
  };

  async plant(opts: ImportPodOpts<TConfig>): Promise<void> {
    const cleanConfig = this.cleanConfig(opts.config);
    await this.prepare(opts);
    await this.execute(cleanConfig);
  }

  async fetch() {
    const podsDir = path.join(this.opts.wsRoot, "pods");
    const podPath = getPodPath(podsDir, OgAWSPod);
    const repoPath = path.join(podPath, "repo");
    if (!fs.existsSync(repoPath)) {
      const repo = new GitRepo({
        localUrl: repoPath,
        remoteUrl: "https://github.com/open-guides/og-aws.git",
      });
      await repo.checkout();
    } else {
      // throw Error("pull not implemented");
    }
    return repoPath;
  }

  async execute(_opts: { src: URI }) {
    const ctx = "OgAWSPod";
    this.L.info({ ctx, msg: "enter" });
    const fetchedSrc = await this.fetch();
    this.L.info({ ctx, msg: "done fetching" });
    const fpath = path.join(fetchedSrc, "README.md");
    const dataPath = fs.readFileSync(fpath, { encoding: "utf8" });
    remark()
      .use(markdownParse, { gfm: true })
      .use(ogAWSPlugin)
      .processSync(dataPath);
    const output = getOutput();
    fs.writeJSONSync("/tmp/out.json", output, { spaces: 4 });

    const dataSplit = dataPath.split("\n");
    await Promise.all(
      // skip last one, high availability
      output.slice(0, -1).map(async (ent, index) => {
        const { name, payload } = ent;
        let start = payload.position?.start.line as number;
        start += 1;
        let end = output[index + 1]?.payload.position?.start.line || -1;
        end -= 1;
        const content = [`# ${name}`].concat(dataSplit.slice(start, end));
        const cleanName = cleanFileName(name.toLowerCase()).replace(",","");
        const fname = `s.${cleanName}`;
        const n = new Note({ id: cleanName, created: "0", updated: "0", fname, body: content.join("\n") });
        console.log(fname);
        return this.engine.write(n, {
          newNode: true,
        });
      })
    );
  }
}
