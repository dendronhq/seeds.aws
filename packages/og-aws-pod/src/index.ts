import { OgAWSPod } from "./importPod";


const out = {
  importPod: OgAWSPod
};

export default out;
