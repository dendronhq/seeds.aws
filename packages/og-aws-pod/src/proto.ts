import path from "path";
import { OgAWSPod } from "./importPod";

async function main() {
  console.log("start");
  const wsRoot= "/Users/kevinlin/projects/dendronv2/dendron-seedbank/vaults/aws";
  const vault = path.join(wsRoot, "vault");
  const pod = new OgAWSPod({roots: [vault], wsRoot})
  await pod.plant({mode: "notes", config: {src: ""}});
  console.log("done");
}

main();
