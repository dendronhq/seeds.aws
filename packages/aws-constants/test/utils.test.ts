import {
  normalizeServiceName,
  awsAliasToFull,
  awsSuffixToFullServiceName,
  validateAWSServiceName
} from "../src/utils";
import { ServiceNames } from "../src";
import * as _ from 'lodash'

describe("awsAliasToFull", () => {
  test("emr", () => {
    expect(awsAliasToFull("EMR")).toEqual("Elastic MapReduce");
  });
  test("CLB", () => {
    expect(awsAliasToFull("CLB")).toEqual("Classic Load Balancer");
  });
  test("s3", () => {
    expect(awsAliasToFull("s3")).toEqual("Simple Storage Service");
  });
  test("exapnd inside normalize", () => {
    expect(normalizeServiceName("EMR")).toEqual("Elastic MapReduce");
  });
  test("exapnd inside normalize 2", () => {
    expect(normalizeServiceName("QLDB")).toEqual("Quantum Ledger Database");
  });
});

describe("normalizeServiceName", () => {
  test.each([
    ["default", {}, "AWS CloudFormation"],
    [
      "no sp, no snake",
      { stripPrefix: false, snakeCase: false },
      "AWS CloudFormation"
    ],
    [
      "no sp, yes snake",
      { stripPrefix: false, snakeCase: true },
      "AWS_CLOUD_FORMATION"
    ],
    [
      "yes sp, no snake",
      { stripPrefix: true, snakeCase: false },
      "CloudFormation"
    ],
    [
      "yes sp, yes snake",
      { stripPrefix: true, snakeCase: true },
      "CLOUD_FORMATION"
    ],
    [
      "no sp, no snake, yes slug",
      { stripPrefix: false, snakeCase: false, slugName: true },
      "AWS CloudFormation"
    ],
    [
      "no sp, yes snake, yes slug",
      { stripPrefix: false, snakeCase: true, slugName: true },
      "AWS_CLOUDFORMATION"
    ],
    [
      "yes sp, no snake, yes slug",
      { stripPrefix: true, snakeCase: false, slugName: true },
      "CloudFormation"
    ],
    [
      "yes sp, yes snake, yes slug",
      { stripPrefix: true, snakeCase: true, slugName: true },
      "CLOUDFORMATION"
    ]
  ])(
    "with prefix, %s, options: %p",
    // @ts-ignore
    (_desc, opts, expected) => {
      expect(
        normalizeServiceName(ServiceNames.AWS_CLOUD_FORMATION, opts)
      ).toEqual(expected);
    }
  );

  test.each([
    ["default", {}, "Service Quotas"],
    [
      "no sp, no snake",
      { stripPrefix: false, snakeCase: false },
      "Service Quotas"
    ],
    [
      "no sp, yes snake",
      { stripPrefix: false, snakeCase: true },
      "SERVICE_QUOTAS"
    ],
    [
      "yes sp, no snake",
      { stripPrefix: true, snakeCase: false },
      "Service Quotas"
    ],
    [
      "yes sp, yes snake",
      { stripPrefix: true, snakeCase: true },
      "SERVICE_QUOTAS"
    ],
    [
      "no sp, no snake, yes slug",
      { stripPrefix: false, snakeCase: false, slugName: true },
      "ServiceQuotas"
    ],
    [
      "no sp, yes snake, yes slug",
      { stripPrefix: false, snakeCase: true, slugName: true },
      "SERVICEQUOTAS"
    ],
    [
      "yes sp, no snake, yes slug",
      { stripPrefix: true, snakeCase: false, slugName: true },
      "ServiceQuotas"
    ],
    [
      "yes sp, yes snake, yes slug",
      { stripPrefix: true, snakeCase: true, slugName: true },
      "SERVICEQUOTAS"
    ]
  ])(
    "no prefix, %s, options: %p",
    // @ts-ignore
    (_desc, opts, expected) => {
      expect(normalizeServiceName(ServiceNames.SERVICE_QUOTAS, opts)).toEqual(
        expected
      );
    }
  );

  test.each([
    ["default", {}, "AWS Cloud Development Kit"],
    [
      "no sp, no snake, no paren",
      { stripPrefix: false, snakeCase: false, stripParen: true },
      "AWS Cloud Development Kit"
    ]
  ])(
    "no prefix, %s, options: %p",
    // @ts-ignore
    (_desc, opts, expected) => {
      expect(
        normalizeServiceName("AWS Cloud Development Kit (AWS CDK)", opts)
      ).toEqual(expected);
    }
  );

  test.each([
    ["default", {}, "Amazon Elastic MapReduce"],
    [
      "no sp, no snake, expand alias",
      { stripPrefix: false, snakeCase: false, expandAlias: true },
      "Amazon Elastic MapReduce"
    ]
  ])(
    "no prefix, %s, options: %p",
    // @ts-ignore
    (_desc, opts, expected) => {
      expect(normalizeServiceName("Amazon EMR", opts)).toEqual(expected);
    }
  );

  test("clb", () => {
    expect(
      normalizeServiceName("clb", {stripPrefix: true})
    ).toEqual("Classic Load Balancer");
  });

  test("capitalize", () => {
    expect(
      normalizeServiceName("Amazon Quicksight", { snakeCase: false })
    ).toEqual("Amazon QuickSight");
  });

  test("strip paren", () => {
    expect(
      normalizeServiceName("Amazon Elastic Block Store (EBS)", {})
    ).toEqual("Amazon Elastic Block Store");
  });

  test("strip paren, snake case", () => {
    expect(
      normalizeServiceName("Amazon Elastic Block Store (EBS)", {
        snakeCase: true
      })
    ).toEqual("AMAZON_ELASTIC_BLOCK_STORE");
  });

  test("aws glacier", () => {
    expect(normalizeServiceName("Amazon Glacier", {})).toEqual(
      "Amazon S3 Glacier"
    );
  });

  test("all service names", () => {
    // @ts-ignore
    expect(_.values(ServiceNames).map( (s: string) => (normalizeServiceName(s, {})))).toMatchSnapshot()
  });
});

describe("awsSuffixToFullServiceName", () => {
  test("normal suffix", () => {
    expect(awsSuffixToFullServiceName("Wavelength")).toEqual(
      "Amazon Wavelength"
    );
  });
  test("suffix with mult entities", () => {
    expect(awsSuffixToFullServiceName("S3 Glacier")).toEqual(
      "Amazon S3 Glacier"
    );
  });
  test("suffix with mult entities 2", () => {
    expect(awsSuffixToFullServiceName("Quantum Ledger Database")).toEqual(
      "Amazon Quantum Ledger Database"
    );
  });
});

describe("validateAWSServiceName", () => {
  test("validate", () => {
    expect(
      validateAWSServiceName("Amazon Quantum Ledger Database")
    ).toBeTruthy();
  });
});
