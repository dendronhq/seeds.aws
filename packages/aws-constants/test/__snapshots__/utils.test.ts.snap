// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeServiceName all service names 1`] = `
Array [
  "AWS Data Exchange",
  "Amazon Elastic Block Store",
  "Amazon Elastic MapReduce",
  "Amazon CloudSearch",
  "AWS Data Pipeline",
  "Amazon Kinesis",
  "Amazon Elasticsearch Service",
  "Amazon QuickSight",
  "Amazon Athena",
  "AWS Glue",
  "AWS Lake Formation",
  "Amazon Managed Streaming for Kafka",
  "Amazon Simple Queue Service",
  "Amazon Simple Notification Service",
  "AWS Step Functions",
  "Amazon MQ",
  "Amazon Sumerian",
  "AWS Cost Explorer",
  "AWS Budgets",
  "Amazon Managed Blockchain",
  "Amazon WorkMail",
  "Amazon Chime",
  "Alexa for Business",
  "Amazon EC2",
  "Amazon Braket",
  "Amazon Wavelength",
  "AWS Elastic Beanstalk",
  "Amazon Elastic Container Service",
  "AWS Lambda",
  "Amazon Elastic Container Registry",
  "Amazon Lightsail",
  "AWS Batch",
  "AWS Serverless Application Repository",
  "Amazon Elastic Container Service for Kubernetes",
  "AWS Outposts",
  "Amazon Simple Email Service",
  "Amazon Mobile Analytics",
  "Amazon Connect",
  "Amazon SimpleDB",
  "Amazon ElastiCache",
  "Amazon DynamoDB",
  "Amazon Redshift",
  "Amazon RDS",
  "Amazon Neptune",
  "Amazon Quantum Ledger Database",
  "Amazon Managed Apache Cassandra Service",
  "Amazon Timestream",
  "Amazon DocumentDB",
  "AWS CodeDeploy",
  "AWS CodePipeline",
  "AWS CodeCommit",
  "AWS CodeBuild",
  "AWS CodeStar",
  "AWS X-Ray",
  "AWS Cloud9",
  "Amazon Appstream",
  "Amazon WorkSpaces",
  "Amazon WorkDocs",
  "Amazon AppStream 2.0",
  "Amazon WorkLink",
  "Amazon GameLift",
  "AWS IoT Core",
  "AWS Greengrass",
  "AWS IoT Device Management",
  "Amazon FreeRTOS",
  "AWS IoT Analytics",
  "AWS IoT 1-Click",
  "AWS IoT Device Defender",
  "AWS IoT SiteWise",
  "AWS IoT Events",
  "AWS IoT Things Graph",
  "Amazon Polly",
  "Amazon Rekognition",
  "Amazon Lex",
  "Amazon Comprehend",
  "Amazon SageMaker",
  "Amazon Fraud Detector",
  "Amazon Kendra",
  "Amazon Transcribe",
  "Amazon Translate",
  "AWS DeepLens",
  "AWS DeepComposer",
  "Amazon Elastic Inference",
  "Amazon Forecast",
  "Amazon Inferentia",
  "AWS DeepRacer",
  "Amazon Textract",
  "Amazon Personalize",
  "AWS CloudFormation",
  "AWS Trusted Advisor",
  "AWS OpsWorks",
  "AWS CloudTrail",
  "AWS Config",
  "AWS Service Catalog",
  "AWS Personal Health Dashboard",
  "AWS Systems Manager",
  "AWS Managed Services",
  "AWS Organizations",
  "AWS Auto Scaling",
  "AWS License Manager",
  "AWS Well-Architected Tool",
  "Amazon CloudWatch",
  "AWS Control Tower",
  "Amazon Elastic Transcoder",
  "AWS Elemental MediaConvert",
  "AWS Elemental MediaPackage",
  "AWS Elemental MediaLive",
  "AWS Elemental MediaStore",
  "AWS Elemental MediaTailor",
  "Amazon Kinesis Video Streams",
  "AWS Elemental MediaConnect",
  "AWS Snowball Family",
  "AWS Database Migration Service",
  "AWS Application Discovery Service",
  "AWS Server Migration Service",
  "AWS Migration Hub",
  "AWS DataSync",
  "AWS Transfer for SFTP",
  "AWS Device Farm",
  "Amazon Pinpoint",
  "AWS Amplify",
  "AWS AppSync",
  "Amazon CloudFront",
  "Elastic Load Balancing",
  "Amazon Route 53",
  "AWS Direct Connect",
  "Amazon VPC",
  "Amazon API Gateway",
  "AWS Global Accelerator",
  "AWS Cloud Map",
  "AWS App Mesh",
  "AWS RoboMaker",
  "AWS Ground Station",
  "Amazon Detective",
  "AWS Identity & Access Management",
  "AWS CloudHSM",
  "Amazon Cognito",
  "AWS Directory Service",
  "AWS Key Management Service",
  "AWS WAF",
  "AWS Certificate Manager",
  "Amazon Inspector",
  "AWS Shield",
  "AWS Artifact",
  "Amazon Macie",
  "Amazon GuardDuty",
  "AWS Single Sign-On",
  "AWS Secrets Manager",
  "AWS Security Hub",
  "Amazon Simple Storage Service",
  "Amazon S3 Glacier",
  "AWS Storage Gateway",
  "Amazon Elastic File System",
  "Amazon FSx for Lustre",
  "Amazon FSx for Windows File Server",
  "AWS Backup",
  "Amazon Event Bridge",
  "Service Quotas",
  "Amazon Lumberyard",
  "AWS Cloud Development Kit",
  "Application Load Balancer",
  "Classic Load Balancer",
]
`;
